jetty-client/src/main/java/org/eclipse/jetty/client/HttpClient.java

```diff
@@ -972,6 +975,25 @@ public void setDispatchIO(boolean dispatchIO)
     {
     }
 
+    /**
+     * Gets the http compliance mode for parsing http responses.
+     * The default http compliance level is {@link HttpCompliance#RFC7230} which is the latest HTTP/1.1 specification
+     */
+    public HttpCompliance getHttpCompliance()
+    {
+        return httpCompliance;
+    }
+
+    /**
+     * Sets the http compliance mode for parsing http responses.
+     * This affect how weak the {@link HttpParser} parses http responses and which http protocol level is supported
+     * @param httpCompliance The compliance level which is used to actually parse http responses
+     */
+    public void setHttpCompliance(HttpCompliance httpCompliance)
```

Comment:
This is a new API for HttpClient, can't we do this correctly here?
A `Collection<HttpComplianceSection>` instead?

Reply:
`HttpCompliance` is-a `Collection<HttpComplianceSection>`, it is just a really clunky non-dynamic one.    The future is for HttpCompliance to be a class, so this API is future correct.

Reply:
I don't follow.

I thought the `HttpCompliance` enum was a convenient location for pre-defined (immutable?)  `Collection<HttpComplianceSection>` baselines.  Something you can start with then customize.

Reply:
This PR changes `HttpCompliance` from being a single valued enum, to being an enum that contains a Collection<HttpComplianceSection>.   So when you pass a HttpCompliance you get two things: a name (instance) for the mode you are running in AND a `Collection<HttpComplianceSection>`.

This is all good.... except because it is an enum we can only have a precompiled fixed number of such modes.  So we have had to create the custom modes and make them mutable.

In future, we will not be passing around raw `Collection<HttpComplianceSection>` as we will lose the name of the mode.   We will still pass around `HttpCompliance`s instances, it is just that they will be classes rather than enums, so users can define their own.   There will still be `HttpCompliance.RFC7230` constants, but they will be static instances rather than enums - thus the code will be compile compatible, but not runtime compatible.